### ISSUE TITLE
Some locales don't appear in Nightly builds

### ISSUE THREAD

**Issue by Delphine** [mozilla-mobile/fenix#26964]
As soon as a locale starts working on Pontoon, the localization should be imported into Fenix Nightly builds (I remember there has been automation in place for this for a while now, but can't find the specific issue anymore).
However, I see for example that currently, "ann" and "mix" locales are missing.
Can someone help look into this? thanks
CC @JohanLorenzo CC @escapewindow

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21557)

**Comment by escapewindow** [comment]
Tl;dr: it looks like we need to update https://github.com/mozilla-mobile/fenix/blob/main/l10n-release.toml .

- [automation/taskcluster/l10n/locales.py defines RELEASE_LOCALES from l10n-release.toml](https://github.com/mozilla-mobile/fenix/blob/29e871086b277292384344c622e5eb615cba2c22/automation/taskcluster/l10n/locales.py#L43-L69)
- but surely we don't use `automation/taskcluster/` any more, right?
- oh, the [build transform uses automation/taskcluster/l10n/filter-release-translations.py](https://github.com/mozilla-mobile/fenix/blob/29e871086b277292384344c622e5eb615cba2c22/taskcluster/fenix_taskgraph/transforms/build.py#L190-L199)
- which [uses RELEASE_LOCALES](https://github.com/mozilla-mobile/fenix/blob/29e871086b277292384344c622e5eb615cba2c22/automation/taskcluster/l10n/filter-release-translations.py#L34-L35)
- -> we have to keep l10n-releases.toml up to date

**Comment by AaronMT** [comment]
> * but surely we don't use `automation/taskcluster/` any more, right?

`androidTest` is used for UI tests, but this directory can live anywhere in the repo (if this ends up being a clean-up) -> path update in [kind.yml](https://github.com/mozilla-mobile/fenix/blob/main/taskcluster/ci/ui-test/kind.yml)

**Comment by JohanLorenzo** [comment]
@Delphine, I just had a chat with @escapewindow. Just to be more explicit: can someone on the l10n team update `[l10n-release.toml](https://github.com/mozilla-mobile/fenix/blob/af1fbd5eeb4a60677724423e1fed55044c608a15/l10n-release.toml)`? On Releng's end we don't know what locales should be shipped or not 🙂

**Comment by Delphine** [comment]
I can do that for sure.
But I thought there was some kind of automation in place at the time, that could pull from our repo as soon as a locale started working on Pontoon. But maybe I'm just mistaken? We can also chat at All-Hands about this :)

**Comment by JohanLorenzo** [comment]
@Delphine, @pascalchevrel and I just sat down and looked into this issue. There are actually 2 bugs here. 

## 1. `l10n-release.toml` has never been updated automatically

Since https://github.com/mozilla-mobile/fenix/pull/12786, `l10n.toml` was split into 2: `l10n.toml` and `l10n-release.toml`. The former is now used by Nightly only. The original intent was described in https://github.com/mozilla-mobile/fenix/issues/12759. We have some automation that automatically synchronize `l10n.toml` in this current repo with that file in that other repo[1]. The thing is: we never updated this automation to bump `l10n-release.toml`. That's why it's always been bumped manually[2].

@Delphine told me the process has changed since the split. We're now okay to let all new locales ride trains. That's why @Delphine changed this comment[3]. This means we don't have to live with `l10-release.toml` anymore, we can just get rid of it. 

## 2. `ann` and `mix` are still missing from nightly anyway

Like said above, `l10n.toml` is used by Nightly and keeps getting updated automatically. So why aren't strings baked in the nightly builds? @pascalchevrel pointed out the problem right away. Although, both locales are part of `l10n.toml`, strings are actually missing! See [4], there is no `values-ann` nor `values-mix` folder. 

Therefore, we bump `l10n.toml` with locales that have 0% of completion. The 3 of us came up with the idea of implementing a lint task that would turn red if locales are missing. This way, we have a way to warn people about this. 

[1] https://github.com/mozilla-mobile/fenix/blob/82e86a8c54cba71b29b87fc81d813f45288d3cb0/l10n.toml
[2] https://github.com/mozilla-mobile/fenix/commits/39c193929d3459388813b6c65744a978de887b4b/l10n-release.toml
[3] https://github.com/mozilla-mobile/fenix/pull/27113/files#diff-37488fc16313aaabe5128aab60c2a8bfb4b29455b4ff93ef6be68e6268391489R2
[4] https://github.com/mozilla-l10n/android-l10n/tree/281ba0fa1935a26c58024ea63522f7ee28774154/mozilla-mobile/fenix/app/src/main/res

**Comment by rvandermeulen** [comment]
Do we need a similar bug for AC and/or Focus?

**Comment by Delphine** [comment]
I had started filing one for Focus and then got caught up in everything All-Hands. I opened an issue for Focus Android. Not sure how a-c worked locale-wise, I feel like that was different and dependent on Fenix. But not sure at what point. Would be great to have someone look into that ;) thanks all!

**Comment by Delphine** [comment]
As I'm ready to add new locales to Fenix, I was reminded of this open issue.
Looking at the last comments (and correct me if I'm omitting anything), these were the next steps: 
* We don't have to live with l10-release.toml anymore, we can just get rid of it.
* Let's implement a lint task that would turn red if locales are missing. This way, we have a way to warn people about this.
Johan, Aki: how confident are you that we can act upon these some time in the near future? thank you!

**Comment by escapewindow** [comment]
I can bring this up in our Tuesday team meeting if you don’t get an answer before then.

**Comment by Delphine** [comment]
I just recently added two locales to Pontoon: "fur" and "kaa" and I see they have locale files and both are up on Nightly builds. So this part is resolved.
Now I will monitor and see if these locales effectively ride the trains.

**Comment by boek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813863

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
